languages/js/src/Oso.ts

```diff
@@ -319,28 +314,20 @@ export class Oso<
    * @param resourceCls Object type.
    * @returns An array of authorized resources.
    */
-  async authorizedResources<T extends Resource>(
+  async authorizedResources(
     actor: Actor,
     action: Action,
-    resourceCls: Class<T>
-  ): Promise<T[]> {
+    resourceCls: Class<Resource>
+  ): Promise<Resource[]> {
     const query = await this.authorizedQuery(actor, action, resourceCls);
     if (!query) return [];
-    const userType = this.getHost().getType(resourceCls);
-    if (userType === undefined)
-      throw new UnregisteredClassError(resourceCls.name);
-    return (await userType.execQuery(query)) as T[];
+    return await this.getHost().adapter.executeQuery(query);
```

Comment:
Probably don't need this `await` since `Adapter::executeQuery` returns `Promise<Resource[]>`
```suggestion
    return this.getHost().adapter.executeQuery(query);
```